providers/hetznerv2/hetznerV2Provider.go

```diff
@@ -0,0 +1,258 @@
+				TTL:  &ttl,
+			}
+			for _, r := range instruction.New {
+				opts.Records = append(opts.Records, hcloud.ZoneRRSetRecord{
+					Value: r.GetTargetCombinedFunc(txtutil.EncodeQuoted),
+				})
+			}
+			reports = append(reports, &models.Correction{
+				F: func() error {
+					result, _, err2 := h.client.Zone.CreateRRSet(context.Background(), z, opts)
+					if err2 != nil {
+						return err2
+					}
+					return h.client.Action.WaitFor(context.Background(), result.Action)
+				},
+				Msg: instruction.MsgsJoined,
+			})
+		case diff2.CHANGE:
+			rrSet := instruction.Old[0].Original.(*hcloud.ZoneRRSet)
+			reports = append(reports, &models.Correction{
+				F: func() error {
+					if instruction.New[0].TTL != instruction.Old[0].TTL {
+						ttl := int(instruction.New[0].TTL)
+						opts := hcloud.ZoneRRSetChangeTTLOpts{TTL: &ttl}
+						action, _, err2 := h.client.Zone.ChangeRRSetTTL(context.Background(), rrSet, opts)
+						if err2 != nil {
+							return err2
+						}
+						err2 = h.client.Action.WaitFor(context.Background(), action)
+						if err2 != nil {
+							return err2
+						}
+					}
+
+					opts := hcloud.ZoneRRSetSetRecordsOpts{}
+					for _, r := range instruction.New {
+						opts.Records = append(opts.Records, hcloud.ZoneRRSetRecord{
+							Value: r.GetTargetCombinedFunc(txtutil.EncodeQuoted),
+						})
+					}
+					action, _, err2 := h.client.Zone.SetRRSetRecords(context.Background(), rrSet, opts)
+					if err2 != nil {
+						return err2
+					}
+					return h.client.Action.WaitFor(context.Background(), action)
+				},
+				Msg: instruction.MsgsJoined,
+			})
+		case diff2.DELETE:
+			reports = append(reports, &models.Correction{
+				F: func() error {
+					rc := instruction.Old[0].Original.(*hcloud.ZoneRRSet)
+					result, _, err2 := h.client.Zone.DeleteRRSet(context.Background(), rc)
+					if err2 != nil {
+						return err2
+					}
+					return h.client.Action.WaitFor(context.Background(), result.Action)
+				},
+				Msg: instruction.MsgsJoined,
+			})
+		}
+	}
+
+	return reports, actualChangeCount, nil
+}
+
+// GetNameservers returns the nameservers for a domain.
+func (h *hetznerV2Provider) GetNameservers(domain string) ([]*models.Nameserver, error) {
+	z, err := h.zoneCache.GetZone(domain)
+	if err != nil {
+		return nil, err
+	}
+	return models.ToNameserversStripTD(z.AuthoritativeNameservers.Assigned)
+}
+
+// GetZoneRecords gets the records of a zone and returns them in RecordConfig format.
+func (h *hetznerV2Provider) GetZoneRecords(domain string, _ map[string]string) (models.Records, error) {
+	z, err := h.zoneCache.GetZone(domain)
+	if err != nil {
+		return nil, err
+	}
+	records, err := h.client.Zone.AllRRSets(context.Background(), z)
+	if err != nil {
+		return nil, err
+	}
+	existingRecords := make([]*models.RecordConfig, 0, len(records))
+	for _, rrSet := range records {
+		if rrSet.Type == hcloud.ZoneRRSetTypeSOA {
+			// SOA records are not available for editing, hide them.
+			continue
+		}
```

Comment:
You should be able to update a SOA records, but the creation and deletion is managed by the API.

So you could skip the creation and deletion in `GetZoneRecordsCorrections` and only apply updates?